app_dart/bin/server.dart

```diff
@@ -82,23 +82,119 @@ Future<void> main() async {
         authProvider,
         scheduler,
       ),
+
+      /// Updates cocoon task status.
+      ///
+      /// This API updates task status in datastore.
+      ///
+      /// POST: /api-update-status
+      ///
+      /// Parameters:
+      ///   CommitBranch: (string in body). Branch of commit.
+      ///   CommitSha: (string in body). Sha of commit.
+      ///   BuilderName: (string in body). Name of the luci builder.
+      ///   NewStatus: (string in body) required. Status of the task.
+      ///   ResultData: (string in body) optional. Benchmark data.
+      ///   BenchmarkScoreKeys: (string in body) optional. Benchmark data.
+      ///
+      /// Response: Status 200 OK
+      ///
+      /// Requires authentication: Status: 401 Unauthorized
       '/api/update-task-status': UpdateTaskStatus(config, swarmingAuthProvider),
       '/api/vacuum-github-commits': VacuumGithubCommits(
         config,
         authProvider,
         scheduler: scheduler,
       ),
+
+      /// Returns status of the tree.
+      ///
+      /// Returns serialized proto with enum representing the
+      /// status of the tree and list of offending tasks.
+      ///
+      /// GET: /api/public/build-status
+      ///
+      /// Parameters:
+      ///   branch: (string in query) default: 'master'. Name of the repo branch.
+      ///
+      /// Response: Status 200 OK
+      ///  {
+      ///    1: 2,
+      ///    2: [ "win_tool_tests_commands", "win_build_test", "win_module_test"]
+      ///   }
       '/api/public/build-status': CacheRequestHandler<Body>(
         cache: cache,
         config: config,
         delegate: GetBuildStatus(config),
         ttl: const Duration(seconds: 15),
       ),
+
+      /// Returns task results for commits.
+      ///
+      /// Returns result details about each task in each checklist for every commit.
+      ///
+      /// GET: /api/public/get-status
+      ///
+      /// Parameters:
+      ///   branch: (string in query) default: 'master'. Name of the branch to filter on.
+      ///   repo: (string in query) default: 'flutter/flutter'. Name of repo to filter on.
```

Comment:
Diff in this particular file are a carry over from another PR. The only change specific to this PR is addition of this "repo:" line to the doc.